ACE/ace/DLL_Manager.cpp

```diff
@@ -177,11 +177,8 @@ ACE_DLL_Handle::open (const ACE_TCHAR *dll_name,
               if (ACE_TString::npos != name->strstr (ACE_TEXT (".a")))
                 {
                   ACE_TCHAR aix_pathname[MAXPATHLEN + 1];
-                  ACE_OS::strncpy (aix_pathname,
-                                   name->c_str (),
-                                   name->length ());
-                  aix_pathname[name->length ()] = '\0';
-                  ACE_OS::strcat (aix_pathname, ACE_TEXT ("(shr.o)"));
+                  ACE_OS::snprintf (aix_pathname, sizeof (aix_pathname),
+                                    ACE_TEXT ("%s(shr.o)"), name->c_str ());
```

Comment:
Shouldn't the return value of ACE_OS::snprintf be checked?

Reply:
Strictly speaking, yes. However, the code is better than it was before. Alternately the combined length could be checked before attempting ACE_OS:snprintf() because the lengths are fixed and known.